File: client/lib/features/items/models/item_model.dart
class ItemModel {
  const ItemModel({
    required this.id,
    required this.name,
    required this.unitPrice,
    required this.unit,
    required this.category,
    this.isActive = true,
    this.vendorId,
  });

  final String id;
  final String name;
  final double unitPrice;
  final String unit;
  final String category;
  final bool isActive;
  final String? vendorId;

  factory ItemModel.fromJson(Map<String, dynamic> json) {
    final id = json['_id']?.toString() ?? json['id']?.toString() ?? '';
    return ItemModel(
      id: id,
      name: json['name']?.toString() ?? '',
      unitPrice: (json['unitPrice'] is num)
          ? (json['unitPrice'] as num).toDouble()
          : 0,
      unit: json['unit']?.toString() ?? 'piece',
      category: json['category']?.toString() ?? 'other',
      isActive: json['isActive'] as bool? ?? true,
      vendorId: json['vendorId']?.toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'unitPrice': unitPrice,
      'unit': unit,
      'category': category,
      'isActive': isActive,
    };
  }
}

File: server/models/Item.model.js
import mongoose from "mongoose";

const itemSchema = new mongoose.Schema(
  {
    ownerId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    name: {
      type: String,
      required: true,
      trim: true,
    },
    unitPrice: {
      type: Number,
      required: true,
      min: 0,
    },
    unit: {
      type: String,
      enum: ["piece", "bowl", "plate", "glass", "other"],
      default: "piece",
    },
    category: {
      type: String,
      trim: true,
      default: "",
    },
    /** Vegetarian vs non-vegetarian — used for order filtering and kitchen prep. */
    dietType: {
      type: String,
      enum: ["veg", "non_veg"],
      default: "veg",
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  {
    timestamps: true,
  }
);

itemSchema.index({ ownerId: 1, isActive: 1 });
itemSchema.index({ ownerId: 1, name: 1 });

const Item = mongoose.model("Item", itemSchema);

export default Item;
